.github/workflows/main.yml

```diff
@@ -97,14 +97,14 @@ jobs:
 
       - name: Checkout branch code
         uses: actions/checkout@v4
-        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name }}
+        if: ${{ github.event_name != 'workflow_dispatch' && github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name }}
```

Comment:
real intended event_name is pull_request_target, changed to explicit negation to avoid skips in case of errors